src/charm.py

```diff
@@ -51,39 +49,159 @@ def __init__(self, *args: Any) -> None:
             self.on.cos_agent_relation_departed, self._on_cos_agent_relation_departed
         )
 
-        self._stored.set_default(config={}, exporter_installed=False, installed=False)
+        self._stored.set_default(
+            config={},
+            exporter_installed=False,
+            installed=False,
+            hw_white_list=[],
+            missing_resources=[],
+            failed_strategies=[],
+            installed_strategies=[],
+        )
         self.num_cos_agent_relations = self.get_num_cos_agent_relations("cos-agent")
 
+    def _log_stored(self) -> None:
+        """Logger print _stored information."""
+        msg = (
+            "Hw tool state: "  # type: ignore
+            f"hw_white_list: {list(self._stored.hw_white_list)}"
+            f" missing_resources: {list(self._stored.missing_resources)}"
+            f" installed_strategies: {list(self._stored.installed_strategies)}"
+            f" failed_strategies: {list(self._stored.failed_strategies)}"
+        )
+        logger.info(msg)
+
+    def str_list_to_hwtool_list(self, values: List[str]) -> List[HWTool]:
+        """Convent list of enum's value to enum."""
+        return [HWTool(value) for value in values]
+
+    @property
+    def state_resource_installed(self) -> bool:
+        """Return True if resources installed completely."""
+        if (
+            len(self._stored.missing_resources) > 0  # type: ignore
+            or len(self._stored.failed_strategies) > 0  # type: ignore
+        ):
+            return False
+        return True
+
+    @property
+    def state_missing_resource(self) -> bool:
+        """Return True if charm is missing resources."""
+        if len(self._stored.missing_resources) == 0:  # type: ignore
+            return False
+        return True
```

Comment:
Maybe this would be equivalent?
```suggestion
        return len(self._stored.missing_resources) > 0
```